clean up warnings given by -Wextra

Both gcc and Clang have a ```-Wextra``` flag, for even more warnings than ```-Wall```, but it's quite a lot of warnings, so I'm leaving this here as a note to get back to it.

Here's what turned up, excluding gmock-all.cc and using ```-Wno-unused-parameter```:

```
In file included from cpp/log/file_db_cert.cc:1:
./cpp/log/file_db-inl.h:320:23: error: comparison of integers of different signs: 'int64_t' (aka 'long') and 'size_type' (aka 'unsigned long')
      [-Werror,-Wsign-compare]
  if (sequence_number < dense_entries_.size()) {
      ~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~
cpp/log/file_db_cert.cc:4:16: note: in instantiation of member function 'FileDB<cert_trans::LoggedCertificate>::HashFromIndex' requested here
template class FileDB<cert_trans::LoggedCertificate>;
               ^
In file included from cpp/log/file_db_cert.cc:1:
./cpp/log/file_db-inl.h:344:23: error: comparison of integers of different signs: 'int64_t' (aka 'long') and 'size_type' (aka 'unsigned long')
      [-Werror,-Wsign-compare]
  if (sequence_number == dense_entries_.size()) {
      ~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~
cpp/log/file_db_cert.cc:4:16: note: in instantiation of member function 'FileDB<cert_trans::LoggedCertificate>::InsertEntryMapping' requested here
template class FileDB<cert_trans::LoggedCertificate>;
               ^
In file included from cpp/log/file_db_cert.cc:1:
./cpp/log/file_db-inl.h:354:21: error: comparison of integers of different signs: 'const long' and 'size_type' (aka 'unsigned long') [-Werror,-Wsign-compare]
      if (it->first != dense_entries_.size()) {
          ~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~
In file included from cpp/log/log_lookup_cert.cc:1:
./cpp/log/log_lookup-inl.h:53:25: error: comparison of integers of different signs: '::google::protobuf::int64' (aka 'long') and 'size_t'
      (aka 'unsigned long') [-Werror,-Wsign-compare]
        sth.tree_size() >= cert_tree_.LeafCount())
        ~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~
In file included from cpp/log/sqlite_db_cert.cc:2:
./cpp/log/sqlite_db-inl.h:503:18: error: comparison of integers of different signs: 'const sqlite3_uint64' (aka 'const unsigned long long') and 'int64_t'
      (aka 'long') [-Werror,-Wsign-compare]
    if (sequence != tree_size_) {
        ~~~~~~~~ ^  ~~~~~~~~~~
In file included from cpp/log/tree_signer_cert.cc:2:
./cpp/log/tree_signer-inl.h:291:24: error: comparison of integers of different signs: 'size_t' (aka 'unsigned long') and '::google::protobuf::int64'
      (aka 'long') [-Werror,-Wsign-compare]
  for (size_t i = 0; i < sth.tree_size(); ++i) {
                     ~ ^ ~~~~~~~~~~~~~~~
```

--- a/cpp/client/ct.cc
+++ b/cpp/client/ct.cc
@@ -249,8 +249,7 @@ static bool PrecertChainToEntry(const cert_trans::PreCertChain& chain,
 }
 
 static bool VerifySCTAndPopulateSSLClientCTData(
-    const SignedCertificateTimestamp& sct, const LogEntry& log_entry,
-    SSLClientCTData* ct_data) {
+    const SignedCertificateTimestamp& sct, SSLClientCTData* ct_data) {
   SSLClientCTData::SCTInfo* sct_info = ct_data->add_attached_sct_info();
   sct_info->mutable_sct()->CopyFrom(sct);
   LogVerifier* verifier = GetLogVerifierFromFlags();
@@ -277,7 +276,7 @@ static bool CheckSCT(const SignedCertificateTimestamp& sct,
     return false;
   }
   ct_data->mutable_reconstructed_entry()->CopyFrom(entry);
-  return VerifySCTAndPopulateSSLClientCTData(sct, entry, ct_data);
+  return VerifySCTAndPopulateSSLClientCTData(sct, ct_data);
 }
 
 // Checks an SCT issued for a Precert.
@@ -289,7 +288,7 @@ static bool CheckSCT(const SignedCertificateTimestamp& sct,
     return false;
   }
   ct_data->mutable_reconstructed_entry()->CopyFrom(entry);
-  return VerifySCTAndPopulateSSLClientCTData(sct, entry, ct_data);
+  return VerifySCTAndPopulateSSLClientCTData(sct, ct_data);
 }
 
 void WriteFile(const std::string& file, const std::string& contents,

--- a/cpp/client/ssl_client.cc
+++ b/cpp/client/ssl_client.cc
@@ -24,9 +24,9 @@ using std::string;
 const uint16_t CT_EXTENSION_TYPE = 18;
 
 // static
-int SSLClient::ExtensionCallback(SSL* s, unsigned ext_type,
-                                 const unsigned char* in, size_t inlen,
-                                 int* al, void* arg) {
+int SSLClient::ExtensionCallback(SSL*, unsigned ext_type,
+                                 const unsigned char* in, size_t inlen, int*,
+                                 void* arg) {
   char pem_name[100];
   unsigned char ext_buf[4 + 65536];
 

--- a/cpp/fetcher/remote_peer.cc
+++ b/cpp/fetcher/remote_peer.cc
@@ -10,6 +10,7 @@
 
 using cert_trans::LoggedCertificate;
 using ct::SignedTreeHead;
+using std::bind;
 using std::chrono::seconds;
 using std::lock_guard;
 using std::make_shared;
@@ -56,13 +57,13 @@ struct RemotePeer::Impl {
   mutex lock_;
   shared_ptr<SignedTreeHead> sth_;
 
-  void FetchSTH(util::Task* task);
+  void FetchSTH();
   void DoneGetSTH(const std::shared_ptr<ct::SignedTreeHead>& on_new_sth,
                   AsyncLogClient::Status status);
 };
 
 
-void RemotePeer::Impl::FetchSTH(util::Task* task) {
+void RemotePeer::Impl::FetchSTH() {
   if (CHECK_NOTNULL(task_)->CancelRequested()) {
     task_->Return(util::Status::CANCELLED);
     return;
@@ -80,69 +81,74 @@ void RemotePeer::Impl::DoneGetSTH(
     return;
   }
 
-  bool sth_provisionally_valid(true);
+  if (status == AsyncLogClient::OK) {
+    bool sth_provisionally_valid(true);
 
-  const LogVerifier::VerifyResult result(verifier_->VerifySignedTreeHead(
-      *new_sth, 0, (time(NULL) + 10) * kNumMillisPerSecond));
+    const LogVerifier::VerifyResult result(verifier_->VerifySignedTreeHead(
+        *new_sth, 0, (time(NULL) + 10) * kNumMillisPerSecond));
 
-  // TODO(alcutter): We should probably log these invalid STHs somewhere a bit
-  // more durable.
-  switch (result) {
-    case LogVerifier::VERIFY_OK:
-      // Alright!
-      break;
-    case LogVerifier::INVALID_TIMESTAMP:
-      LOG(WARNING) << "Invalid timestamp on received STH:\n"
-                   << new_sth->DebugString();
-      invalid_sths_received->Increment("invalid_timestamp");
+    // TODO(alcutter): We should probably log these invalid STHs somewhere a bit
+    // more durable.
+    switch (result) {
+      case LogVerifier::VERIFY_OK:
+        // Alright!
+        break;
+      case LogVerifier::INVALID_TIMESTAMP:
+        LOG(WARNING) << "Invalid timestamp on received STH:\n"
+                     << new_sth->DebugString();
+        invalid_sths_received->Increment("invalid_timestamp");
+        sth_provisionally_valid = false;
+        break;
+      case LogVerifier::INVALID_SIGNATURE:
+        LOG(WARNING) << "Invalid signature on received STH:\n"
+                     << new_sth->DebugString();
+        invalid_sths_received->Increment("invalid_signature");
+        sth_provisionally_valid = false;
+        break;
+      case LogVerifier::INCONSISTENT_TIMESTAMPS:
+      case LogVerifier::INVALID_MERKLE_PATH:
+      case LogVerifier::INVALID_FORMAT:
+        sth_provisionally_valid = false;
+        LOG(FATAL) << "Unexpected verify result for VerifySignedTreeHead(): "
+                   << result;
+    }
+
+    const string local_root_at_snapshot(
+        log_lookup_->RootAtSnapshot(new_sth->tree_size()));
+    if (new_sth->sha256_root_hash() != local_root_at_snapshot) {
+      LOG(WARNING) << "Received STH:\n" << new_sth->DebugString()
+                   << " whose root:\n" << HexString(new_sth->sha256_root_hash())
+                   << "\ndoes not match that of local tree at corresponding "
+                   << "snapshot:\n" << HexString(local_root_at_snapshot);
+      invalid_sths_received->Increment("incorrect_root");
       sth_provisionally_valid = false;
-      break;
-    case LogVerifier::INVALID_SIGNATURE:
-      LOG(WARNING) << "Invalid signature on received STH:\n"
-                   << new_sth->DebugString();
-      invalid_sths_received->Increment("invalid_signature");
-      sth_provisionally_valid = false;
-      break;
-    case LogVerifier::INCONSISTENT_TIMESTAMPS:
-    case LogVerifier::INVALID_MERKLE_PATH:
-    case LogVerifier::INVALID_FORMAT:
-      sth_provisionally_valid = false;
-      LOG(FATAL) << "Unexpected verify result for VerifySignedTreeHead(): "
-                 << result;
-  }
+    }
 
-  const string local_root_at_snapshot(
-      log_lookup_->RootAtSnapshot(new_sth->tree_size()));
-  if (new_sth->sha256_root_hash() != local_root_at_snapshot) {
-    LOG(WARNING) << "Received STH:\n" << new_sth->DebugString()
-                 << " whose root:\n" << HexString(new_sth->sha256_root_hash())
-                 << "\ndoes not match that of local tree at corresponding "
-                 << "snapshot:\n" << HexString(local_root_at_snapshot);
-    invalid_sths_received->Increment("incorrect_root");
-    sth_provisionally_valid = false;
-  }
+    if (sth_provisionally_valid) {
+      lock_guard<mutex> lock(lock_);
 
-  if (sth_provisionally_valid) {
-    lock_guard<mutex> lock(lock_);
-
-    if (new_sth->tree_size() < 0) {
-      LOG(WARNING) << "Unexpected tree size in new_sth:\n"
-                   << new_sth->DebugString();
-    } else if (sth_ && new_sth->tree_size() < sth_->tree_size()) {
-      LOG(WARNING) << "Received old STH:\n" << new_sth->DebugString();
-    } else if (!sth_ || new_sth->timestamp() > sth_->timestamp()) {
-      // This STH is good, we'll take it.
-      sth_ = new_sth;
-      if (on_new_sth_) {
-        on_new_sth_(*sth_);
+      if (new_sth->tree_size() < 0) {
+        LOG(WARNING) << "Unexpected tree size in new_sth:\n"
+                     << new_sth->DebugString();
+      } else if (sth_ && new_sth->tree_size() < sth_->tree_size()) {
+        LOG(WARNING) << "Received old STH:\n" << new_sth->DebugString();
+      } else if (!sth_ || new_sth->timestamp() > sth_->timestamp()) {
+        // This STH is good, we'll take it.
+        sth_ = new_sth;
+        if (on_new_sth_) {
+          on_new_sth_(*sth_);
+        }
       }
     }
+  } else {
+    LOG(WARNING) << "Problem fetching STH, got error " << status
+                 << " from AsyncLogClient";
   }
 
   // Schedule another STH fetch
   task_->executor()->Delay(
       seconds(FLAGS_remote_peer_sth_refresh_interval_seconds),
-      task_->AddChild(bind(&RemotePeer::Impl::FetchSTH, this, _1)));
+      task_->AddChild(bind(&RemotePeer::Impl::FetchSTH, this)));
 }
 
 
@@ -157,7 +163,7 @@ RemotePeer::RemotePeer(
   TaskHold hold(task);
   task->DeleteWhenDone(impl_);
 
-  impl_->FetchSTH(nullptr);
+  impl_->FetchSTH();
 }
 
 

--- a/cpp/log/ct_extensions.cc
+++ b/cpp/log/ct_extensions.cc
@@ -40,8 +40,7 @@ static const char kPrecertificateSigningLN[] =
 static const char kASN1NullValue[] = "NULL";
 
 // String conversion for an ASN1 NULL
-static char* ASN1NullToString(X509V3_EXT_METHOD* method,
-                              ASN1_NULL* asn1_null) {
+static char* ASN1NullToString(X509V3_EXT_METHOD*, ASN1_NULL* asn1_null) {
   if (asn1_null == NULL)
     return NULL;
   char* buf = strdup(kASN1NullValue);
@@ -49,7 +48,7 @@ static char* ASN1NullToString(X509V3_EXT_METHOD* method,
 }
 
 // String conversion from an ASN1:NULL conf.
-static ASN1_NULL* StringToASN1Null(X509V3_EXT_METHOD* method, X509V3_CTX* ctx,
+static ASN1_NULL* StringToASN1Null(X509V3_EXT_METHOD*, X509V3_CTX*,
                                    char* str) {
   if (str == NULL || strcmp(str, kASN1NullValue) != 0) {
     return NULL;

--- a/cpp/log/leveldb_db-inl.h
+++ b/cpp/log/leveldb_db-inl.h
@@ -69,7 +69,7 @@ int64_t KeyToIndex(leveldb::Slice key) {
 
   int64_t index(0);
   CHECK_EQ(index_str.size(), sizeof(index));
-  for (int i = 0; i < sizeof(index); ++i) {
+  for (size_t i = 0; i < sizeof(index); ++i) {
     index = (index << 8) | static_cast<unsigned char>(index_str[i]);
   }
 

--- a/cpp/log/log_lookup-inl.h
+++ b/cpp/log/log_lookup-inl.h
@@ -49,8 +49,9 @@ void LogLookup<Logged>::UpdateFromSTH(const ct::SignedTreeHead& sth) {
   if (sth.timestamp() == latest_tree_head_.timestamp())
     return;
 
+  CHECK_LE(0, sth.tree_size());
   if (sth.timestamp() <= latest_tree_head_.timestamp() ||
-      sth.tree_size() < cert_tree_.LeafCount()) {
+      static_cast<uint64_t>(sth.tree_size()) < cert_tree_.LeafCount()) {
     LOG(WARNING) << "Database replied with an STH that is older than ours: "
                  << "Our STH:\n" << latest_tree_head_.DebugString()
                  << "Database STH:\n" << sth.DebugString();

--- a/cpp/log/sqlite_db-inl.h
+++ b/cpp/log/sqlite_db-inl.h
@@ -423,7 +423,7 @@ int64_t SQLiteDB<Logged>::TreeSize() const {
   while (ret == SQLITE_ROW) {
     const sqlite3_uint64 sequence(statement.GetUInt64(0));
 
-    if (sequence != tree_size_) {
+    if (sequence != static_cast<uint64_t>(tree_size_)) {
       return tree_size_;
     }
 

--- a/cpp/monitoring/gcm/exporter.cc
+++ b/cpp/monitoring/gcm/exporter.cc
@@ -25,6 +25,7 @@ DEFINE_int32(google_compute_monitoring_credentials_refresh_interval_minutes,
 
 namespace cert_trans {
 
+using std::bind;
 using std::chrono::minutes;
 using std::chrono::seconds;
 using std::chrono::system_clock;
@@ -67,7 +68,7 @@ GCMExporter::GCMExporter(const string& instance_name, UrlFetcher* fetcher,
   CreateMetrics();
   // Push the first set of metrics, this will also schedule the next push
   // automatically.
-  PushMetrics(nullptr /* we're not a Task here */);
+  PushMetrics();
 }
 
 
@@ -189,7 +190,7 @@ void AddLabel(const string& key, const string& value, JsonObject* labels) {
 }  // namespace
 
 
-void GCMExporter::PushMetrics(util::Task* child_task) {
+void GCMExporter::PushMetrics() {
   if (task_.task()->CancelRequested()) {
     task_.task()->Return(util::Status::CANCELLED);
     return;
@@ -215,7 +216,7 @@ void GCMExporter::PushMetrics(util::Task* child_task) {
     CHECK_NOTNULL(m);
     for (auto& p : m->CurrentValues()) {
       JsonObject labels;
-      for (int i(0); i < p.first.size(); ++i) {
+      for (size_t i(0); i < p.first.size(); ++i) {
         AddLabel(m->LabelName(i), p.first[i], &labels);
       }
 
@@ -264,7 +265,7 @@ void GCMExporter::PushMetrics(util::Task* child_task) {
 
   executor_->Delay(
       seconds(FLAGS_google_compute_monitoring_push_interval_seconds),
-      task_.task()->AddChild(bind(&GCMExporter::PushMetrics, this, _1)));
+      task_.task()->AddChild(bind(&GCMExporter::PushMetrics, this)));
 }
 
 

--- a/cpp/monitoring/gcm/exporter.h
+++ b/cpp/monitoring/gcm/exporter.h
@@ -22,7 +22,7 @@ class GCMExporter {
  private:
   void RefreshCredentials();
   void CreateMetrics() const;
-  void PushMetrics(util::Task* child_task);
+  void PushMetrics();
 
   const std::string instance_name_;
   UrlFetcher* const fetcher_;

--- a/cpp/monitoring/labelled_values.h
+++ b/cpp/monitoring/labelled_values.h
@@ -47,7 +47,7 @@ struct i__ {};
 
 
 template <class Tuple>
-void label_values(const Tuple& t, std::vector<std::string>* values, i__<0>) {
+void label_values(const Tuple&, std::vector<std::string>*, i__<0>) {
 }
 
 

--- a/cpp/monitoring/prometheus/exporter.cc
+++ b/cpp/monitoring/prometheus/exporter.cc
@@ -18,7 +18,7 @@ void AddLabelTypes(::io::prometheus::client::Metric* metric,
                    const std::vector<std::string>& values) {
   CHECK_NOTNULL(metric);
   CHECK_EQ(names.size(), values.size());
-  for (int i(0); i < names.size(); ++i) {
+  for (size_t i(0); i < names.size(); ++i) {
     ::io::prometheus::client::LabelPair* label_pair(metric->add_label());
     label_pair->set_name(names[i]);
     label_pair->set_value(values[i]);

--- a/cpp/net/connection_pool.cc
+++ b/cpp/net/connection_pool.cc
@@ -188,7 +188,7 @@ ConnectionPool::ConnectionPool(libevent::Base* base)
 
 // static
 evhtp_res ConnectionPool::Connection::ConnectionErrorHook(
-    evhtp_connection_t* conn, evhtp_error_flags errtype, void* arg) {
+    evhtp_connection_t* conn, evhtp_error_flags, void* arg) {
   CHECK_NOTNULL(conn);
   CHECK_NOTNULL(arg);
   CHECK(libevent::Base::OnEventThread());

--- a/cpp/server/ct-dns-server.cc
+++ b/cpp/server/ct-dns-server.cc
@@ -22,18 +22,14 @@ DEFINE_string(domain, "", "Domain");
 DEFINE_string(db, "", "Database for certificate and tree storage");
 
 // Basic sanity checks on flag values.
-static bool ValidatePort(const char* flagname, int32_t port) {
-  if (port <= 0 || port > 65535) {
-    std::cerr << "Port value " << port << " is invalid. " << std::endl;
-    return false;
-  }
-  return true;
+static bool ValidatePort(const char*, int32_t port) {
+  return (port <= 0 || port > 65535);
 }
 
 static const bool port_dummy =
     RegisterFlagValidator(&FLAGS_port, &ValidatePort);
 
-static bool NonEmptyString(const char* flagname, const string& str) {
+static bool NonEmptyString(const char*, const string& str) {
   return !str.empty();
 }
 

--- a/cpp/server/ct-mirror.cc
+++ b/cpp/server/ct-mirror.cc
@@ -148,7 +148,7 @@ Gauge<>* latest_local_tree_size_gauge =
 
 
 // Basic sanity checks on flag values.
-static bool ValidatePort(const char* flagname, int port) {
+static bool ValidatePort(const char*, int port) {
   if (port <= 0 || port > 65535) {
     std::cout << "Port value " << port << " is invalid. " << std::endl;
     return false;
@@ -372,7 +372,7 @@ int main(int argc, char* argv[]) {
                           new MerkleVerifier(new Sha256Hasher))),
       server.log_lookup(), new_sth,
       fetcher_task.task()->AddChild(
-          [](Task* task) { LOG(INFO) << "RemotePeer exited."; })));
+          [](Task*) { LOG(INFO) << "RemotePeer exited."; })));
   const unique_ptr<ContinuousFetcher> fetcher(
       ContinuousFetcher::New(event_base.get(), &pool, db, false));
   fetcher->AddPeer("target", peer);
@@ -381,9 +381,8 @@ int main(int argc, char* argv[]) {
 
   thread sth_updater(&STHUpdater, db, server.cluster_state_controller(),
                      &queue_mutex, &queue,
-                     fetcher_task.task()->AddChild([](Task* task) {
-                       LOG(INFO) << "STHUpdater exited.";
-                     }));
+                     fetcher_task.task()->AddChild(
+                         [](Task*) { LOG(INFO) << "STHUpdater exited."; }));
 
   server.Run();
 

--- a/cpp/server/ct-server.cc
+++ b/cpp/server/ct-server.cc
@@ -148,7 +148,7 @@ Latency<milliseconds> signer_run_latency_ms("signer_run_latency_ms",
 
 
 // Basic sanity checks on flag values.
-static bool ValidatePort(const char* flagname, int port) {
+static bool ValidatePort(const char*, int port) {
   if (port <= 0 || port > 65535) {
     std::cout << "Port value " << port << " is invalid. " << std::endl;
     return false;

--- a/cpp/server/handler.cc
+++ b/cpp/server/handler.cc
@@ -237,7 +237,7 @@ HttpHandler::HttpHandler(
       node_is_stale_(controller_->NodeIsStale()) {
   event_base_->Delay(seconds(FLAGS_staleness_check_delay_secs),
                      task_.task()->AddChild(
-                         bind(&HttpHandler::UpdateNodeStaleness, this, _1)));
+                         bind(&HttpHandler::UpdateNodeStaleness, this)));
 }
 
 
@@ -568,7 +568,7 @@ bool HttpHandler::IsNodeStale() const {
 }
 
 
-void HttpHandler::UpdateNodeStaleness(util::Task* task) {
+void HttpHandler::UpdateNodeStaleness() {
   if (!task_.task()->IsActive()) {
     // We're shutting down, just return.
     return;
@@ -582,5 +582,5 @@ void HttpHandler::UpdateNodeStaleness(util::Task* task) {
 
   event_base_->Delay(seconds(FLAGS_staleness_check_delay_secs),
                      task_.task()->AddChild(
-                         bind(&HttpHandler::UpdateNodeStaleness, this, _1)));
+                         bind(&HttpHandler::UpdateNodeStaleness, this)));
 }

--- a/cpp/server/handler.h
+++ b/cpp/server/handler.h
@@ -70,7 +70,7 @@ class HttpHandler {
                            const std::shared_ptr<PreCertChain>& chain) const;
 
   bool IsNodeStale() const;
-  void UpdateNodeStaleness(util::Task* task);
+  void UpdateNodeStaleness();
 
   JsonOutput* const output_;
   LogLookup<LoggedCertificate>* const log_lookup_;

--- a/cpp/server/server.h
+++ b/cpp/server/server.h
@@ -148,7 +148,7 @@ void RefreshNodeState(ClusterStateController<LoggedCertificate>* controller,
 }
 
 
-void WatchdogTimeout(int sig) {
+void WatchdogTimeout(int) {
   if (FLAGS_watchdog_timeout_is_fatal) {
     LOG(FATAL) << "Watchdog timed out, killing process.";
   } else {

--- a/cpp/util/etcd.cc
+++ b/cpp/util/etcd.cc
@@ -479,7 +479,7 @@ void EtcdClient::WatchInitialGetDone(WatchState* state, GetResponse* resp,
                  << " second(s)";
     state->task_->executor()->Delay(
         seconds(FLAGS_etcd_watch_error_retry_delay_seconds),
-        state->task_->AddChild([this, state](Task* task) {
+        state->task_->AddChild([this, state](Task*) {
           this->WatchRequestDone(state, nullptr, nullptr);
         }));
     return;

--- a/cpp/util/libevent_wrapper.cc
+++ b/cpp/util/libevent_wrapper.cc
@@ -37,7 +37,7 @@ void FreeEvDns(evdns_base* dns) {
 }
 
 
-static void Handler_ExitLoop(evutil_socket_t sig, short events, void* base) {
+static void Handler_ExitLoop(evutil_socket_t, short, void* base) {
   event_base_loopexit((event_base*)base, NULL);
 }
 
@@ -56,7 +56,7 @@ void DelayCancel(event* timer, util::Task* task) {
 }
 
 
-void DelayDispatch(evutil_socket_t sock, short events, void* userdata) {
+void DelayDispatch(evutil_socket_t, short, void* userdata) {
   static_cast<util::Task*>(CHECK_NOTNULL(userdata))->Return();
 }
 
@@ -284,7 +284,7 @@ evhtp_connection_t* Base::HttpsConnectionNew(const string& host,
 }
 
 
-void Base::RunClosures(evutil_socket_t sock, short flag, void* userdata) {
+void Base::RunClosures(evutil_socket_t, short, void* userdata) {
   Base* self(static_cast<Base*>(CHECK_NOTNULL(userdata)));
 
   vector<function<void()>> closures;

--- a/cpp/util/masterelection.cc
+++ b/cpp/util/masterelection.cc
@@ -349,13 +349,12 @@ void MasterElection::UpdateProposal(const string& backed) {
                          seconds(FLAGS_master_keepalive_interval_seconds * 2),
                          my_proposal_modified_index_, resp,
                          new Task(bind(&MasterElection::ProposalUpdateDone,
-                                       this, backed, resp, _1),
+                                       this, resp, _1),
                                   base_.get()));
 }
 
 
-void MasterElection::ProposalUpdateDone(const string& backed,
-                                        EtcdClient::Response* resp,
+void MasterElection::ProposalUpdateDone(EtcdClient::Response* resp,
                                         Task* task) {
   unique_ptr<EtcdClient::Response> resp_deleter(resp);
   unique_lock<mutex> lock(mutex_);

--- a/cpp/util/masterelection.h
+++ b/cpp/util/masterelection.h
@@ -105,8 +105,7 @@ class MasterElection {
   void UpdateProposal(const std::string& backed);
 
   // Called when our proposal file has been refreshed by the KeepAlive thread.
-  void ProposalUpdateDone(const std::string& backed,
-                          EtcdClient::Response* resp, util::Task* task);
+  void ProposalUpdateDone(EtcdClient::Response* resp, util::Task* task);
 
   // Deletes this node's proposal.
   // This should only be called on the base_ event thread.

